src/pages/library/SearchAndFilterComponent.test.ts

```diff
@@ -0,0 +1,65 @@
+import * as _ from 'lodash/fp'
+import {
+  FilterSection,
+  getMatchingDataForSection,
+  getMatchingDataForSectionList,
+  listItemsMatchForSectionEntry, sectionEntrySelected
+} from 'src/pages/library/SearchAndFilterComponent'
+
+
+const data = ['A quick brown fox jumped over a lazy gray dog', 'aeio', 'stock', 'styx', 'xylophone']
+
+// Arrange
+const FILTERS: FilterSection<string>[] = [
+  {
+    header: 'Contains vowel',
+    matchBy: (listValue, sectionEntry) => _.includes(sectionEntry, listValue),
+    values: ['a', 'e', 'i', 'o', 'u']
+  },
+  {
+    header: 'Starts with',
+    matchBy: (listValue, sectionEntry) => _.startsWith(sectionEntry, listValue),
+    values: ['x', 'y']
+  }
+]
+
+describe('library/common', () => {
+  it('matches correctly for a section entry', () => {
+    // Act
+    const listItemsMatching = listItemsMatchForSectionEntry('a', FILTERS[0].matchBy, data)
+    // Assert
+    expect(listItemsMatching).toContain(data[0])
+    expect(listItemsMatching).toContain(data[1])
+    expect(listItemsMatching).not.toContain(data[2])
+    expect(listItemsMatching).not.toContain(data[3])
+    expect(listItemsMatching).not.toContain(data[4])
```

Comment:
Would it be worth making `data` an enum so that it's clear which of these items is being tested across these tests? For example `expect(listItemsMatching).not.toContain(data.STOCK)`